diff --git a/generators/chipyard/src/main/resources/csrc/emulator.cc b/generators/chipyard/src/main/resources/csrc/emulator.cc
--- a/generators/chipyard/src/main/resources/csrc/emulator.cc
+++ b/generators/chipyard/src/main/resources/csrc/emulator.cc
@@ -387,6 +387,7 @@
 
+  if (tile) tile->final();
   if (dtm) delete dtm;
   if (tsi) delete tsi;
   if (jtag) delete jtag;
   if (tile) delete tile;
   if (verilated_argv) delete[] verilated_argv;
