diff --git a/build.sbt b/build.sbt
index 099b5d9..fa14cc4 100644
--- a/build.sbt
+++ b/build.sbt
@@ -27,7 +27,7 @@ def javacOptionsVersion(scalaVersion: String): Seq[String] = {
 }
 
 // Provide a managed dependency on X if -DXVersion="" is supplied on the command line.
-val defaultVersions = Map("firrtl" -> "1.5-SNAPSHOT")
+val defaultVersions = Map("firrtl" -> "1.4.1")
 
 lazy val baseSettings = Seq(
   name := "treadle",

