diff --git a/src/grt/src/Grid.h b/src/grt/src/Grid.h
index e6d45a241..3dcda591b 100644
--- a/src/grt/src/Grid.h
+++ b/src/grt/src/Grid.h
@@ -153 +153 @@ class Grid
-  int pitches_in_tile_ = 15;
+  int pitches_in_tile_ = 48;
diff --git a/src/grt/src/fastroute/src/utility.cpp b/src/grt/src/fastroute/src/utility.cpp
index eff586910..e22b7eeb4 100644
--- a/src/grt/src/fastroute/src/utility.cpp
+++ b/src/grt/src/fastroute/src/utility.cpp
@@ -1632 +1632 @@ void FastRouteCore::check2DEdgesUsage()
-  const int max_usage_multiplier = 100;
+  const int max_usage_multiplier = 101;
