diff --git a/src/grt/src/Grid.h b/src/grt/src/Grid.h
index e6d45a241..9bbe7e2c7 100644
--- a/src/grt/src/Grid.h
+++ b/src/grt/src/Grid.h
@@ -150,7 +150,7 @@ class Grid
   bool perfect_regular_y_;
   int num_layers_;
-  int pitches_in_tile_ = 15;
+  int pitches_in_tile_ = 16;
   std::vector<int> track_pitches_;
   std::vector<int> horizontal_edges_capacities_;
   std::vector<int> vertical_edges_capacities_;
 };

 }  // namespace grt
